atomix/cluster/src/main/java/io/atomix/raft/storage/RaftStorage.java

```diff
@@ -73,7 +74,12 @@ private RaftStorage(
     this.persistedSnapshotStore = persistedSnapshotStore;
     this.journalIndexDensity = journalIndexDensity;
 
-    IoUtil.ensureDirectoryExists(directory, prefix + " raft partition storage");
+    try {
+      FileUtil.ensureDirectoryExists(directory.toPath());
+    } catch (final IOException e) {
+      throw new UncheckedIOException(
```

Comment:
Please check with @npepinpe. Afaik he is working on guidelines for exception handling. Last time we talked about the matter he preferred `LangUtil.rethrowUnchecked(e);` over using `UncheckedIOException`

Other than that, I would approve the review.

Reply:
For the purpose of this PR that's fine, especially since we're wrapping explicitly an IOException (so the indirection from UncheckedIOException -> IOException isn't as bad imo as UncheckedIOException -> JsonProcessingException), I don't want to block you. That said, my 2 cents here, I would avoid doing I/O in a constructor if I can, and if I can't, then throw the checked exception. imo anyone calling constructors most likely doesn't think about handling potentially I/O errors most of the time.

Reply:
I guess, moving this check outside of the constructor is out of scope for this issue. Besides, currently there is nothing a caller of this constructor can do to handle this error other than failing the startup.